### ISSUE TITLE
Deleting Dupes - Deleting is very slow

### ISSUE THREAD

**Issue by francwalter** [vaeth/bookmarkdupes#104]
Halo
I have some thousands of bookmarks in my Firefox on Ubuntu, Windows (2 devices), Mac and Android devices.
All with sync enabled.
But for the DupeDeleting I disabled Sync in FF 111.0.1 (Ubuntu 22.04 at the moment).

Many dupes because of wrong sync I guess.
I tried this Add-On to delete the duplicates, but it is incredibly slow.
After half an hour I read:
`(3 %) 74 von 2566 Lesezeichen entfernt`
Is this normal or can I enhance it somehow?

Like this it will be finished tomorrow, slowing down FF as well, so I cannot use it.
Thanks.
frank

EDIT: maybe I could run the SQL command directly on the places.sqlite DB?
Then I would quit FF and run that command it in e.g. DBeaver or on command line, but there is no visible logfile in the add-on, where I could see what is done in background.

Here a screenshot before deleting:

![grafik](https://user-images.githubusercontent.com/927269/227553279-86e0fba9-1d9a-4de0-84ff-4d481d220c8d.png)

Here when I start deleting:

![grafik](https://user-images.githubusercontent.com/927269/227553324-9b6fe16e-506b-4f2e-b3c5-ac2711f8d9ca.png)

And here after about 10 minutes deleting, makes FF very slow:

![grafik](https://user-images.githubusercontent.com/927269/227553335-4a80efe8-c1f8-4cd0-b01d-5af589e39497.png)

**Comment by francwalter** [comment]
Oh oh!
And here, after I disabled some other Add-Ons (Bookmark Notes, Bookmark Search Plus 2 and Bookmark Organizer) only some seconds after starting it:

![grafik](https://user-images.githubusercontent.com/927269/227554407-3707f4ed-9686-4256-bb02-f316a68849d6.png)

Sorry, that was house made :blush: 
Close it.

EDIT: and already finished:

`2447 Lesezeichen gelöscht`

**Comment by vaeth** [comment]
Normally, the deletion is quite fast (in the range of hundreds of bookmarks per seconds).
I had reports about slow deletions which turned out to be caused by completely broken bookmark databases.
Another reason might be lack of memory (frequent garbage collection).

The extension just calls the javascript methods for bookmark deletions - that's all it has permissions to.
Manually accessing the database  with sqllite is probably not a good idea as this will presumably not play well with syncing.

BTW, this might be another reason for the slow processing, if for some reason firefox thinks that it has to sync very frequently.

**Comment by francwalter** [comment]
Sorry, I have closed already, we wrote the same time.
Issue solved, it was obviously the other Bookmark Add-Ons (see above) which made it slow.
Thanks for the fast reply and the great and useful Add-On :heart: 

By the way, my places.sqlite is very crowdy and maybe not optimized and even partly broken. I had for a long time big issues with sync on Mac but since a while sync works again on Mac, I guess something serverside was fixed to handle errors or such.
But I have too much bookmarks to clean them manually.

A big thing was, when Moz stopped the description field in bookmarks, about FF 60 something, I depended a lot on that.
Never found a replacement for the personal comment field (where I noted a lot).
